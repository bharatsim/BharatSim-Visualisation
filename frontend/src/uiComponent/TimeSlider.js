import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Slider from '@material-ui/core/Slider';
import Typography from '@material-ui/core/Typography';
import PropTypes from 'prop-types';

const useStyles = makeStyles((theme) => ({
  root: {
    height: theme.spacing(15),
  },
}));

export default function TimeSlider({
  defaultValue,
  maxValue,
  step,
  minValue,
  setTimeSliderValue,
  title,
  timeSliderValue,
  dataTestId,
}) {
  const classes = useStyles();

  function onChange(e, value) {
    setTimeSliderValue(value);
  }

  return (
    <div className={classes.root}>
      <Typography variant="body2">{`${title} ${timeSliderValue}`}</Typography>
      <Slider
        defaultValue={defaultValue}
        aria-labelledby="Time Slider"
        step={step}
        marks
        onChange={onChange}
        min={minValue}
        max={maxValue}
        valueLabelDisplay="auto"
        data-testid={dataTestId}
      />
    </div>
  );
}

TimeSlider.defaultProps = {
  dataTestId: '',
};

TimeSlider.propTypes = {
  defaultValue: PropTypes.number.isRequired,
  maxValue: PropTypes.number.isRequired,
  step: PropTypes.number.isRequired,
  minValue: PropTypes.number.isRequired,
  setTimeSliderValue: PropTypes.func.isRequired,
  title: PropTypes.string.isRequired,
  timeSliderValue: PropTypes.number.isRequired,
  dataTestId: PropTypes.string,
};
