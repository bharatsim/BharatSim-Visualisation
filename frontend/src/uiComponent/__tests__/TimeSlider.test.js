import React from 'react';
import { render } from '@testing-library/react';
import { fireEvent } from '@testing-library/dom';
import TimeSlider from '../TimeSlider';

const props = {
  defaultValue: 1,
  maxValue: 5,
  step: 1,
  minValue: 1,
  setTimeSliderValue: jest.fn(),
  title: 'title',
  timeSliderValue: 1,
};
jest.mock('@material-ui/core/Slider', () => (props) => {
  const { id, name, min, max, onChange, testid } = props;
  return (
    <input
      data-testid={testid}
      type="range"
      id={id}
      name={name}
      min={min}
      max={max}
      onChange={(event, val) => onChange(val)}
    />
  );
});
describe('<TimeSlider />', () => {
  it('should match snapshot', () => {
    const { container } = render(<TimeSlider {...props} />);
    expect(container).toMatchSnapshot();
  });
});
