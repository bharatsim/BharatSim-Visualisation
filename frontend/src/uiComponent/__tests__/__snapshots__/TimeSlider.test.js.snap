// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TimeSlider /> should match snapshot 1`] = `
<div>
  <div
    class="makeStyles-root-1"
  >
    <div
      class="MuiBox-root MuiBox-root-3"
    >
      <p
        class="MuiTypography-root makeStyles-title-2 MuiTypography-body1"
      >
        slider title: 1
      </p>
    </div>
    <div>
      <pre>
        defaultValue = 1
        aria-labelledby = Time Slider
        step = null
        marks = [{"value":1,"label":1},{"value":3},{"value":5,"label":5}]
        onChange = function onChange
        min = 1
        max = 5
        value = 1
        valueLabelDisplay = off
      </pre>
      <pre
        data-testid="marks-container"
      >
        [{"value":1,"label":1},{"value":3},{"value":5,"label":5}]
      </pre>
      <input
        aria-labelledby="Time Slider"
        data-testid="time-slider"
        max="5"
        min="1"
        value="1"
      />
    </div>
  </div>
</div>
`;
